packages/shared/src/components/sidebar/Sidebar.tsx

```diff
@@ -67,6 +71,19 @@ export default function Sidebar({
   const showSettings = modal?.type === LazyModal.UserSettings;
   const { isNewMobileLayout } = useMobileUxExperiment();
   const isTablet = useViewSize(ViewSize.Tablet);
+  const sidebarToggle = useFeature(feature.sidebarToggle);
+  const { completeAction, checkHasCompleted } = useActions();
+  const hasChangedToggle = checkHasCompleted(ActionType.SidebarToggle);
+  const sidebarToggleState = hasChangedToggle ? sidebarExpanded : sidebarToggle;
+
+  const onToggleSidebar = () => {
+    if (!sidebarToggle && !hasChangedToggle) {
+      completeAction(ActionType.SidebarToggle);
+      return onSidebarExpanded(true);
+    }
+
+    return toggleSidebarExpanded();
+  };
```

Comment:
Why do we need such a complicated logic? Can we instead change the `SettingsContext` to take the `feature.sidebarToggle` into account?

In this code, I don't really know what is happening :)

Reply:
Yeah, totally agree with that. Hence, I raised the discussion about making it [through the API](https://dailydotdev.slack.com/archives/C02E2C3C13R/p1713449074470439).

From the SettingsContext alone, it won't still know whether the value is coming from the hard-coded defaults or the actual settings. If we go through API, it will all be easier.

Reply:
@sshanzel since this is new users only, what do you think that we just do it in onboarding after user registration. We can call `UPDATE_USER_SETTINGS_MUTATION` there and set the sidebar value depending if user is in experiment. All other logic then stays the same?

Basically it will simplify almost as if we did it on the API side (as you said) but it will be easier to implement I think.

Reply:
That's a good idea, I'm just thinking it would need other implementation if product decides after this they want to experiment with current users, but maybe we don't need to care much at this point as it depends..

Reply:
Good point. We can do that yes. Will try it quickly.